cscs-checks/microbenchmarks/cpu/roofline/berkeley-ert.py

```diff
@@ -0,0 +1,444 @@
+    #if (ERT_FLOP & 64) == 64 /* add 64 flops */
+    #if (ERT_FLOP & 128) == 128 /* add 128 flops */
+    #if (ERT_FLOP & 256) == 256 /* add 256 flops */
+    #if (ERT_FLOP & 512) == 512 /* add 512 flops */
+    #if (ERT_FLOP & 1024) == 1024 /* add 1024 flops */
+
+    A typical output is:
+    working_set_size*bytes_per_elem
+                           t
+                                     seconds  total_bytes  total_flops
+           1024            1         731.238         2048       131072
+           1024            2           7.705         4096       262144
+           1024            4          11.562         8192       524288
+           1024            8          22.242        16384      1048576
+           etc...
+    """
+    # }}}
+    def __init__(self):
+        self.descr = f'Empirical Roofline Toolkit (Base for building/running)'
+        # pe step
+        # build step
+        # run step
+        # postprocess step
+        # sanity step
+
+    # {{{ hooks
+    @rfm.run_before('compile')
+    def set_compiler_flags_and_variables(self):
+        self.sourcesdir = os.path.join(
+            self.current_system.resourcesdir,
+            'roofline',
+            'cs-roofline-toolkit.git',
+            'Empirical_Roofline_Tool-1.1.0',
+        )
+        self.readonly_files = [
+            'Batch',
+            'Config',
+            'Drivers',
+            'ert',
+            'ert_cscs.py',
+            'ERT_Users_Manual.pdf',
+            'Kernels',
+            'Plot',
+            'Python',
+            'README.md',
+            'Results',
+            'Scripts',
+        ]
+        self.sourcepath = 'Kernels/kernel1.cxx Drivers/driver1.cxx'
+        self.build_system = 'SingleSource'
+        # get all parameters:
+        ert_trials_min = sn.getattr(self, 'ert_trials_min')
+        ert_precision = sn.getattr(self, 'ert_precision')
+        ert_flop = sn.getattr(self, 'ert_flop')
+        self.build_system.cppflags = [
+            '-I./Kernels',
+            f'-DERT_FLOP={ert_flop}',
+            '-DERT_ALIGN=32',
+            '-DERT_MEMORY_MAX=1073741824',
+            '-DERT_MPI=True',
+            '-DERT_OPENMP=True',
+            '-DERT_WORKING_SET_MIN=1',
+            '-DERT_WSS_MULT=1.1',
+            f'-D{ert_precision}',
+            f'-DERT_TRIALS_MIN={ert_trials_min}',
+            # keeping as reminder
+            # '-DERT_INTEL',
+        ]
+        self.prgenv_flags = {
+            'PrgEnv-gnu': ['-fopenmp', '-O3'],
+        }
+        self.build_system.cxxflags = \
+            self.prgenv_flags[self.current_environ.name]
+        self.variables = {
+            'OMP_NUM_THREADS': str(self.num_cpus_per_task),
+            'OMP_PROC_BIND': 'close',
+            'OMP_PLACES': 'cores',
+        }
+        self.prebuild_cmds = ['module list', 'which gcc']
+
+    @rfm.run_before('run')
+    def set_run_cmds(self):
+        self.prerun_cmds += [f'for ii in `seq {ert_repeat}`;do']
```

Comment:
This is not very nice, especially, that you have to close the `for` loop in the `postrun_cmds`. Perhaps you can do the for loop in Python and generate the series of `srun` commands that you want. Check here: https://reframe-hpc.readthedocs.io/en/stable/tutorial_advanced.html#adding-more-parallel-launch-commands